File: ts/management-api-v2/cm_api_v2_put_variant_unpublish_archive.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client.unpublishLanguageVariant()
  // .byItemId('f4b3fc05-e988-4dae-9ac1-a94aba566474')
  .byItemCodename('my_article')
  // .byItemExternalId('59713')
  // .byLanguageId('d1f95fde-af02-b3b5-bd9e-f232311ccab8')
  .byLanguageCodename('es-mx')
  // To schedule unpublish date, use .withData({scheduled_to: 'datetime-to-unpublish-at'})
  .withData({
    scheduled_to: '2038-01-19T04:14:08',
    display_timezone: 'Australia/Sydney'
  })
  // To unpublish now, use .withoutData()
  .toPromise();

File: net/management-api-v2/cm_api_v2_put_variant_unpublish_archive.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.LanguageVariants;
using Kontent.Ai.Management.Models.Publishing;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var identifier = LanguageVariantIdentifier.ByIds(Guid.Parse("f4b3fc05-e988-4dae-9ac1-a94aba566474"), Guid.Parse("d1f95fde-af02-b3b5-bd9e-f232311ccab8"));
// var identifier = LanguageVariantIdentifier.ByCodenames("my_article", "es-ES");

// Immediate unpublish
(await client.UnpublishLanguageVariantAsync(identifier)).EnsureSuccess();

// Scheduled unpublish
(await client.ScheduleUnpublishingOfLanguageVariantAsync(identifier, new ScheduleModel
{
    ScheduledTo = new DateTimeOffset(2038, 1, 19, 4, 14, 8, TimeSpan.Zero),
    DisplayTimeZone = "Australia/Sydney"
})).EnsureSuccess();
